src/Sidekick.Business/Apis/Poe/Trade/Data/Stats/StatDataService.cs

```diff
@@ -157,6 +170,17 @@ private void FillMods(List<Modifier> mods, List<StatData> patterns, string text)
                     }
                 }
 
+                if (x.Option != null && x.Option.Options != null && x.Option.Options.Count != 0)
```

Comment:
Could use null propagation here together with the SingleOrDefault call to skip all of these checks.